src/coreclr/jit/importer.cpp

```diff
@@ -9411,6 +9417,9 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                         usesHelper = true;
                         break;
 
+                    case CORINFO_FIELD_STATIC_TLS_MANAGED:
+                        setMethodHaveTlsFieldAccess();
```

Comment:
this might be setting this flag only for inlinee's compiler object and then can be lost during actual inlining, consider
```csharp
[ThreadStatic]
static int f;

static int Foo() => f;

static int Bar() => Foo();
```
check codegen for `Bar`

Reply:
It generates correctly.

```asm
G_M9148_IG01:
       sub      rsp, 40
                                                ;; size=4 bbWeight=1 PerfScore 0.25
G_M9148_IG02:
       mov      rax, qword ptr GS:[0x0058]
       mov      rax, qword ptr [rax+30H]
       mov      ecx, dword ptr [rax+48H]
       cmp      ecx, 2
       jl       SHORT G_M9148_IG06
                                                ;; size=21 bbWeight=1 PerfScore 6.25
G_M9148_IG03:
       mov      rax, qword ptr [rax+50H]
       mov      rdx, qword ptr [rax+10H]
       test     rdx, rdx
       je       SHORT G_M9148_IG06
                                                ;; size=13 bbWeight=0.97 PerfScore 5.09
G_M9148_IG04:
       mov      eax, dword ptr [rdx+8CH]
                                                ;; size=6 bbWeight=1 PerfScore 2.00
G_M9148_IG05:
       add      rsp, 40
       ret
                                                ;; size=5 bbWeight=1 PerfScore 1.25
G_M9148_IG06:
       mov      ecx, 2
       call     CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE_NOCTOR_OPTIMIZED
       mov      rdx, rax
       jmp      SHORT G_M9148_IG04
```